mgmt-lambda/auth.ts

```diff
@@ -0,0 +1,27 @@
+import type { AuthSettings } from './model/AuthSettings'
+import { SecretsManagerClient, GetSecretValueCommand, GetSecretValueResponse } from '@aws-sdk/client-secrets-manager'
+
+export async function getAuthSettings(secretManagerClient: SecretsManagerClient): Promise<AuthSettings> {
+  const secretName = process.env.SettingsSecretName
+  if (!secretName) {
+    throw new Error('Unable to retrieve secret. Error: unable to get secret name')
```

Comment:
I'd suggest to have more clear error message, for instance `getAuthSettings failed: env variable SettingsSecretName not found`.